refactor(examples) Update federated kaplan meier fitter (#3922)
adap/flower: examples/federated-kaplan-meier-fitter/README.md

Reviewer: Need to add `title` to the metadata section. This title should be the same as in the pyprojec.toml. Which should also match with the top title in the README.md

Author: Solved: `Federated Survival Analysis with Flower and KaplanMeierFitter`

<img src="_static/survival_function_federated.png" alt="Survival Function" width="600"/>
</p>

## Set up the project

### Clone the project

Start by cloning the example project:

```shell
$ git clone --depth=1 https://github.com/adap/flower.git _tmp && mv _tmp/examples/federated-kaplan-meier-fitter . && rm -rf _tmp && cd federated-kaplan-meier-fitter
```

This will create a new directory called `federated-kaplan-meier-fitter`  with the following structure:

```shell
federated-kaplan-meier-fitter
├── examplefmk
│   ├── __init__.py
│   ├── client_app.py   # Defines your ClientApp
│   ├── server_app.py   # Defines your ServerApp
│   └── task.py         # Defines your model, training and data loading
├── pyproject.toml      # Project metadata like dependencies and configs
└── README.md
```

### Install dependencies and project

> Install the dependencies defined in `pyproject.toml` as well as the `federated-kaplan-meier-fitter` package.

Reviewer: ```suggestion
Install the dependencies defined in `pyproject.toml` as well as the `examplefmk` package.
```

```bash
pip install -e .
```

## Run the project

You can run your `ClientApp` and `ServerApp` in both _simulation_ and _deployment_ mode without making changes to the code. If you are starting with Flower, we recommend you using the _simulation_ model as it requires fewer components to be launched manually. By default, `flwr run` will make use of the Simulation Engine.

### Run with the Simulation Engine

```bash
flwr run .
```

You can also check that the results match the centralized version.

```shell
$ python3 centralized.py
```

```bash
flwr run . --run-config num-server-rounds=5,learning-rate=0.05
```

### Run with the Deployment Engine

> \[!NOTE\]
> An update to this example will show how to run this Flower application with the Deployment Engine and TLS certificates, or with Docker.

> ## Explanation

Reviewer: ```suggestion
```

Author: Done

> This example is a federated version of the centralized case that can be found
> [here](https://github.com/ml-explore/mlx-examples/tree/main/mnist).

Reviewer: ```suggestion
```

Author: Done
